### PR TITLE
Improve test coverage for EventRegistrantsModal to 100%

### LINKED ISSUE #3680: Improve code coverage for: src/components/EventRegistrantsModal/Modal/EventRegistrantsModal.tsx
# Goal  
The goal of this issue is to improve the code coverage for this file

# Tasks  
1. Review the file to identify sections of code that are being ignored by Codecov or are not covered by tests.  
2. Create or update test cases to ensure 100% coverage for the file.  
3. Remove any `/* istanbul ignore */` or equivalent statements that bypass code coverage reporting, unless absolutely necessary.  

# Resources  
1. Refer to the foundational documentation on writing test cases in the repository.  
1. Check the [Codecov report](https://app.codecov.io/gh/PalisadoesFoundation/talawa-admin/tree/develop-postgres?displayType=list) for details on the uncovered lines.  

# Acceptance Criteria  
1. All sections of the file are covered by tests.  
1. Code coverage for the file reaches 100%.  
1. PR created with necessary updates, passing all checks and reviews.
1. All tests must be valid

# Other
If you are new to testing, refer to [past test code coverage improvement PRs](https://github.com/PalisadoesFoundation/talawa-admin/issues?q=is%3Aissue%20state%3Amerged%20in%3Atitle%20test%20)

### PR DESCRIPTION
**Pull request by rudra-sah00**
**What kind of change does this PR introduce?**

Test coverage improvement

**Issue Number:**

Fixes #3680

**Snapshots/Videos:**

Screenshot 2025-10-10 at 12 42 15 PM

**If relevant, did you update the documentation?**

N/A - Test coverage improvement only

**Summary**

This PR improves test coverage for the `EventRegistrantsModal` component to achieve 100% coverage across all metrics (statements, branches, functions, and lines).

**Changes made:**
- Added comprehensive test cases to cover all code paths in `EventRegistrantsModal.tsx`
- Fixed mock data structure to include both `id` and `_id` fields for proper Apollo Client normalization
- Added `__typename` fields to all mock GraphQL responses
- Mocked `AddOnSpotAttendee` component to test its integration
- Added tests for:
  - Add attendee mutation success and failure scenarios
  - Delete attendee mutation success and failure scenarios
  - Autocomplete functionality
  - Loading state rendering
  - Modal close functionality
  - AddOnSpotAttendee `reloadMembers` callback
  - Attendee with only `id` field (no `_id`) to cover fallback branches
- Modified `EventRegistrantsModal.tsx` to handle `_id` fallback: `(attendee as any)._id || attendee.id`

**Coverage Results:**
- ✅ **Statements:** 100% (38/38)
- ✅ **Branches:** 100% (12/12)
- ✅ **Functions:** 100% (15/15)
- ✅ **Lines:** 100% (38/38)

**Does this PR introduce a breaking change?**

No

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass

**Other information**

All 9 test cases pass successfully. The implementation follows existing test patterns in the codebase and ensures robust coverage of all component functionality.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- Bug Fixes
  - Registrant handling now supports multiple ID formats (prefers primary ID when present), so registrants display and can be reliably removed.
  - Modal open/close and on-the-spot registration flows behave more consistently.
  - Success messages for add/remove actions are asserted and shown reliably.

- Tests
  - Expanded coverage with mocked loading states, modal flows, add/remove scenarios, autocomplete/registration paths, and edge cases.
  - Assertions now verify concrete UI messages and interaction sequences.

### CODE REVIEW

File: src/components/EventRegistrantsModal/Modal/EventRegistrantsModal.spec.tsx [hunk 1]
```diff
@@ -15,7 +15,29 @@
...
+    show: boolean;
+    handleClose: () => void;
+    reloadMembers: () => void;
+  }) => {
+    return show ? (
+      <div data-testid="add-onspot-modal">
+        <button onClick={handleClose}>Close Modal</button>
+        <button onClick={reloadMembers} data-testid="reload-members-btn">
+          Reload Members
+        </button>
```
[the remark this answers is not in the document]
Comment (rudra-sah00) [reply src/components/EventRegistrantsModal/Modal/EventRegistrantsModal.spec.tsx:36]: thanks for the suggestion i appreciate the best practice recommendation. however since this is a test mock component and the project does not currently using biome linting. the eslint checks are passing without any warning

File: src/components/EventRegistrantsModal/Modal/EventRegistrantsModal.spec.tsx [hunk 2]
```diff
@@ -15,7 +15,29 @@
 import { ToastContainer } from 'react-toastify';
 import { LocalizationProvider } from '@mui/x-date-pickers';
 import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
-import { describe, test, expect, vi } from 'vitest';
+import { describe, test, expect, vi, beforeEach } from 'vitest';
```
[the remark this answers is not in the document]
Comment (rudra-sah00) [reply src/components/EventRegistrantsModal/Modal/EventRegistrantsModal.spec.tsx]: resolved!

File: src/components/EventRegistrantsModal/Modal/EventRegistrantsModal.spec.tsx [hunk 3]
```diff
@@ -306,14 +337,22 @@ describe('Testing Event Registrants Modal', () => {
...
-    fireEvent.click(queryByTestId('CancelIcon') as Element);
+    // Get all CancelIcon elements (MUI Chip delete icons)
+    const cancelIcons = document.querySelectorAll('[data-testid="CancelIcon"]');
+
+    // The first CancelIcon should be the delete icon for the "John Doe" chip
+    // (assuming there are no other cancel icons before it)
+    const deleteIcon = cancelIcons[0];
+    expect(deleteIcon).toBeTruthy();
+
+    fireEvent.click(deleteIcon as Element);
```
[the remark this answers is not in the document]
Comment (rudra-sah00) [reply src/components/EventRegistrantsModal/Modal/EventRegistrantsModal.spec.tsx:349]: thanks for the suggestion  i appreciate the Testing Library best practice recommendation. 
In this case, I intentionally used `querySelectorAll` because the MUI Chip's `CancelIcon` is rendered internally by the library and doesn't have a custom testid that we control - we're querying the MUI component's own testid. The current approach works reliably and the comment explains the intent clearly.

File: src/components/EventRegistrantsModal/Modal/EventRegistrantsModal.tsx [hunk 4]
```diff
@@ -163,8 +163,16 @@ export const EventRegistrantsModal = (props: ModalPropType): JSX.Element => {
...
+                key={
+                  (attendee as InterfaceUser & { _id?: string })._id ||
+                  attendee.id
+                }
+                onDelete={(): void =>
+                  deleteRegistrant(
+                    (attendee as InterfaceUser & { _id?: string })._id ||
+                      attendee.id,
+                  )
+                }
```
[the remark this answers is not in the document]
Comment (rudra-sah00) [reply src/components/EventRegistrantsModal/Modal/EventRegistrantsModal.tsx:175]: thanks Good point about reducing duplication. Since this PR is focused on test coverage and the current code is working well i will keep it as is for now.

**Comment by rudra-sah00** [comment]
@palisadoes please review

File: src/components/EventRegistrantsModal/Modal/EventRegistrantsModal.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (palisadoes) [reply src/components/EventRegistrantsModal/Modal/EventRegistrantsModal.tsx:175]: Please follow Code Rabbit's suggestion. We need better overall code quality

**Comment by palisadoes** [comment]
Please see my comment.

@coderabbitai full review
